crates/nu-engine/src/eval.rs

```diff
@@ -977,16 +1016,7 @@ pub fn eval_block(
                     )
                 }));
 
-            let redirect_stdout = redirect_stdout
-                || (elements_iter.peek().map_or(false, |next_element| {
-                    matches!(
-                        next_element,
-                        PipelineElement::Redirection(_, Redirection::Stdout, _)
-                            | PipelineElement::Redirection(_, Redirection::StdoutAndStderr, _)
-                            | PipelineElement::Expression(..)
-                            | PipelineElement::SeparateRedirection { .. }
-                    )
-                }));
+            let redirect_stdout = redirect_stdout || elements_iter.peek().is_some();
```

Comment:
I assume the previous `matches!` is now handled somewhere else implicitly?

Reply:
Oh, wait!
Thanks for a good catch, I haven't consider it carefully...

Currently we need the redirection to make it work with this:
```nushell
cat a.txt err> /dev/null | lines
```
But it breaks this:
```nushell
def x [] {
    ^echo a.txt err> /dev/null   # it doesn't output a.txt anymore in this pr
    3
}
```
I need to look into this more

Reply:
It's a pity that the new implementation is more complicated :-(